### PR TITLE
Double colon ftw

### PR DESCRIPTION
**Pull request by jackfranklin**
- support double colons
- even added some tests! (will write more tests for pulldown soon)

One thing, when the tests are run (`node tests/tests.js`), it outputs Pulldown's help with the tests output. I've no idea why. Anyone seen this before?

### CODE REVIEW

File: pulldown.js [hunk 1]
```diff
@@ -61,7 +61,7 @@ Pulldown.prototype.help = function () {
   console.log('  Example usage:');
   console.log();
   console.log('    pulldown jquery             # Downloads jQuery');
-  console.log('    pulldown jquery:jq.js       # Downloads jQuery to jq.js');
+  console.log('    pulldown jquery::jq.js       # Downloads jQuery to jq.js');
```
Comment (tgvashworth) [inline pulldown.js]: Bit of an alignment tweak needed here.

File: pulldown.js [hunk 2]
```diff
@@ -96,12 +96,9 @@ Pulldown.prototype.processUserArgs = function(callback) {
...
 Pulldown.prototype.parsePackageArgument = function(searchTerm, callback) {
-  var split = searchTerm.split(":");
-  var outputName;
-  if (split.length > 1) {
-    outputName = _.last(split);
-    searchTerm = _.initial(split).join(':');
-  }
+  var split = searchTerm.split("::");
+  var outputName = split[1];
+  searchTerm = split[0];
```
Comment (tgvashworth) [inline pulldown.js]: Code like this always makes me nervous. There's an outside chance there will be multiple `::` so I think we should take the last as the output name and join the initial ones, as above. Might be over-cautious tho.

File: tests/urls.js [hunk 3]
```diff
@@ -0,0 +1,30 @@
+var assert = require("assert");
+var Pulldown = require("../pulldown");
+var sinon = require("sinon");
+
+
+describe("Pulling down URLs", function() {
+  var pulldown, theSpy;
+  beforeEach(function() {
+    theSpy = sinon.spy();
```
Comment (tgvashworth) [inline tests/urls.js:9]: Sexy.

**Comment by jackfranklin** [comment]
> One thing, when the tests are run (node tests/tests.js), it outputs Pulldown's help with the tests output. I've no idea why. Anyone seen this before?

This is because Pulldown gets executed when we `require` it in the tests, and then this gets run:

```
var pulldown = new Pulldown();
pulldown.init(process.argv.slice(2));
```

Is there a way to get round this?

File: pulldown.js (hunk 2 above)
Comment (jackfranklin) [inline pulldown.js]: If you can describe to me a case with any chance of there being multiple `::`s, I'll agree :P

**Comment by jackfranklin** [comment]
IF there's a preferred library to Mocha btw, I'm all ears. I like Mocha but if there's something else better, I'm down with that.

**Comment by jackfranklin** [comment]
> Is there a way to get round this?

We could do some checking in Pulldown to see if the args contain `node ./tests/tests.js`, in which case `Pulldown.prototype.help = function() {}`, to stop it. Feels icky tho

File: pulldown.js (hunk 2 above)
Comment (tgvashworth) [inline pulldown.js]: Point is that it's valid in URLs. Still, over caution it may be.

**Comment by tgvashworth** [comment]
@jackfranklin Check this out: http://nodejs.org/docs/latest/api/all.html#all_accessing_the_main_module

So only run init with argv if it's the main module.

**Comment by jackfranklin** [comment]
@phuu hot. Will get this sorted this eve.

File: pulldown.js (hunk 2 above)
Comment (jackfranklin) [inline pulldown.js]: Fine, just to keep you happy.

**Pull request merged**